com.unity.shadergraph/Editor/Drawing/PreviewManager.cs

```diff
@@ -243,276 +282,443 @@ public bool HandleGraphChanges()
                 var node = m_Graph.GetNodeFromGuid(edge.inputSlot.nodeGuid);
                 if(node != null)
                 {
-                    m_NodesToUpdate.Add(node);
+                    m_NodesShaderChanged.Add(node);
                     m_RefreshTimedNodes = true;
                 }
             }
 
-            return m_NodesToUpdate.Count > 0;
+            // This seems to only control whether we rebuild colors or not...
```

Comment:
Why would we only want to rebuild colors when a Node Preview's Shader has changed?
Is it because we only want to rebuild them when an edge is added or removed?

Reply:
I'd rather base the color rebuild logic on what it actually cares about, rather than the return values from another function.   :)

Reply:
Good solution, that makes way more sense. Preview should have nothing to do with it. 👍